brain_observatory_analysis/behavior/video_qc/annotation_tools.py

```diff
@@ -0,0 +1,447 @@
+    """
+    syncfile = glob.glob(os.path.join(path, '*.h5'))[0]
+    print('sync file: {}'.format(syncfile))
+    return syncfile
+
+
+def get_sync_dataset(syncfile):
+    """
+    Get the sync dataset from a given sync file. Use get_syncfile to get the sync file.
+
+    Parameters
+    ----------
+    syncfile : str
+        full path to the sync file
+
+    Returns
+    -------
+    sync_dataset : allensdk.brain_observatory.sync_dataset.SyncDataset
+        sync dataset
+    """
+    sync_dataset_object = sync_dataset.Dataset(syncfile)
+    print('successfully loaded sync dataset object')
+    return sync_dataset_object
+
+
+def get_cam_timestamps(sync_dataset, cam_json, account_for_metadata_frame=True):
+    '''
+    Returns the experiment timestamps for the frames recorded in an MVR video
+
+    Parameters:
+    ------------
+    sync_dataset : allensdk.brain_observatory.sync_dataset.SyncDataset
+        sync dataset object from a given experiment, use get_sync_dataset to get this object
+    cam_json : str or full path to json file saved for each camera view. Use get_jsonfile to get this file
+        json file from MVR.
+    account_for_metadata_frame : True/False, if TRUE prepend a NaN to the exposure times for the
+    metadata frame that is prepended to the MVR video
+    '''
+
+    if isinstance(cam_json, str):
+        cam_json_data = read_json(cam_json)
+
+    total_frames_recorded = cam_json_data['RecordingReport']['FramesRecorded']
+
+    cam_label = cam_json_data['RecordingReport']['CameraLabel']
+    sync_line = get_frame_exposure_times(cam_label)
+
+    exposure_times = sync_dataset.get_rising_edges(sync_line, units='seconds')
+
+    lost_frames = extract_lost_frames_from_json(cam_json_data)
+
+    # filter out lost frames from the sync exposure times
+    frame_times = [e for ie, e in enumerate(exposure_times) if ie not in lost_frames]
+
+    # cut off extra exposure times that didn't make it into the video
+    frame_times = frame_times[:total_frames_recorded]  # hopefully this becomes obsolete after MVR stop sequence changes
+
+    # add a NaN to the beginning for the metadata frame
+    if account_for_metadata_frame:
+        frame_times = np.insert(frame_times, 0, np.nan)
+
+    return np.array(frame_times)
+
+
+def plot_DLC_points(frame, points_to_plot, bodyparts=None):
+    """
+    Plots the DLC points on a frame.
+    """
+    if bodyparts is None:
+        bodyparts = points_to_plot['bodyparts'].unique()
+    bodycolors = image_utils.generate_distinct_colors(len(bodyparts))
+    for b, bodypart in enumerate(bodyparts):
+        one_point = points_to_plot[points_to_plot.bodyparts == bodypart]
+        x = one_point[one_point.coords == 'x']['value'].values[0]
+        y = one_point[one_point.coords == 'y']['value'].values[0]
+        cv2.circle(frame, (int(x), int(y)), 5, bodycolors[b], -1)
+
+    return frame
+
+
+def process_video(output_path, mp4file, df_dlc, start_frame=1000, end_frame=2000):
```

Comment:
The point of this function is to plot_DLC_points overlayed on the movie frames. Process_video is a bit vague and seems like it would have a bunch of processing steps. I think a more obvious function name would be easier to remember when trying to recall functions in annotation tools.